### ISSUE TITLE
what is the world_size argument in SerialSampler

### ISSUE THREAD

**Issue by carlbalmer** [astooke/rlpyt#50]
`SerialSampler.initialize` takes the argument `world_size` what does this do and how do you set it?
It seems to have something to do with DistributedDataParallel in agent...

**Comment by renqibing** [comment]
In my view, world size is the number of GPUs used in this setting.

**Comment by renqibing** [comment]
By the way, can you tell me what does bootstrap_timelimit do? Does it correspond with pretraining? I see all of q function policy gradient algorithms set it as true.

**Comment by astooke** [comment]
That's right, world size is relevant when using the `SyncRl` runner, which runs multi-GPU.

**Comment by astooke** [comment]
> By the way, can you tell me what does bootstrap_timelimit do? Does it correspond with pretraining? I see all of q function policy gradient algorithms set it as true.

Yes this is from the original SAC implementation, it makes the algorithm use the next Q value in the Q-learning target when the episodes ends due to timeout.  Normally, when an episode is `done`, the next Q value would not be used, only the reward.  Intuitively, the time limit is an arbitrary implementation decision, but the agent should learn as if it can run forever.  Not to do with pretraining.  Try it with and without, it makes a difference on e.g. HalfCheetah. (good idea to ask in separate thread in the future, so others can find it  :)

**Comment by renqibing** [comment]
> > By the way, can you tell me what does bootstrap_timelimit do? Does it correspond with pretraining? I see all of q function policy gradient algorithms set it as true.
> 
> Yes this is from the original SAC implementation, it makes the algorithm use the next Q value in the Q-learning target when the episodes ends due to timeout. Normally, when an episode is `done`, the next Q value would not be used, only the reward. Intuitively, the time limit is an arbitrary implementation decision, but the agent should learn as if it can run forever. Not to do with pretraining. Try it with and without, it makes a difference on e.g. HalfCheetah. (good idea to ask in separate thread in the future, so others can find it :)

Thanks for your sincere help!
